src/modules/45-dashboards/pages/home/DashboardForm.tsx

```diff
@@ -0,0 +1,163 @@
+ */
+
+import React from 'react'
+import {
+  Button,
+  Container,
+  Formik,
+  FormikForm as Form,
+  FormInput,
+  Heading,
+  Layout,
+  ModalErrorHandler,
+  ModalErrorHandlerBinding,
+  OverlaySpinner,
+  SelectOption
+} from '@wings-software/uicore'
+import { FontVariation } from '@harness/design-system'
+import * as Yup from 'yup'
+import { useParams } from 'react-router-dom'
+import { useStrings } from 'framework/strings'
+import { useGetFolder, ErrorResponse } from 'services/custom-dashboards'
+import type { AccountPathProps } from '@common/interfaces/RouteInterfaces'
+import css from './HomePage.module.scss'
+
+const TAGS_SEPARATOR = ','
+
+interface DashboardFormProps {
+  formData?: DashboardFormRequestProps
+  title: string
+  loading: boolean
+  onComplete: (data: DashboardFormRequestProps) => void
+  modalErrorHandler?: ModalErrorHandlerBinding
+  setModalErrorHandler: (modalErrorHandler: ModalErrorHandlerBinding) => void
+}
+
+export interface DashboardFormRequestProps {
+  description?: string
+  folderId: string
+  name: string
+}
+
+export interface DashboardFormikValues {
+  description: string[]
+  folderId: string
+  name: string
+}
+
+const DashboardForm: React.FC<DashboardFormProps> = ({
+  formData,
+  title,
+  loading,
+  onComplete,
+  modalErrorHandler,
+  setModalErrorHandler
+}) => {
+  const { getString } = useStrings()
+  const { accountId } = useParams<AccountPathProps>()
+  const [isFoldersLoading, setFoldersLoading] = React.useState(true)
+  const [initialValues, setInitialValues] = React.useState<DashboardFormikValues>({
+    folderId: formData?.folderId || '',
+    name: formData?.name || '',
+    description: formData?.description?.split(TAGS_SEPARATOR) || []
+  })
+  const [folderListItems, setFolderListItems] = React.useState<SelectOption[]>([])
+
+  const sharedFolder = {
+    value: getString('dashboards.shared'),
+    label: getString('dashboards.sharedFolderTitle')
+  }
+
+  const { data: foldersList, error } = useGetFolder({
+    queryParams: { accountId: accountId, page: 1, pageSize: 1000 }
+  })
+
+  const isMissingFolderId = (folders: SelectOption[]) =>
+    !folders.some((item: SelectOption) => item.value === formData?.folderId)
+
+  React.useEffect(() => {
+    const errorResponse = error?.data as ErrorResponse
+    if (errorResponse?.responseMessages) {
+      modalErrorHandler?.showDanger(errorResponse.responseMessages)
+      setFoldersLoading(false)
+    }
+  }, [error, modalErrorHandler])
+
+  React.useEffect(() => {
+    if (foldersList?.resource) {
+      const folders = foldersList?.resource?.map((folder: { id: string; name: string }) => {
+        return { value: folder?.id, label: folder?.name }
+      })
+      if (isMissingFolderId(folders)) {
+        setInitialValues({ ...initialValues, folderId: 'shared' })
+      }
+      setFolderListItems([sharedFolder, ...folders])
+      setFoldersLoading(false)
+    }
+  }, [foldersList])
```

Comment:
Codacy has a fix for the issue: [React Hook React.useEffect has missing dependencies: 'initialValues', 'isMissingFolderId', and 'sharedFolder'. Either include them or remove the dependency array. You can also do a functional update 'setInitialValues(i => ...)' if you only need 'initialValues' in the 'setInitialValues' call.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232275)
```suggestion
  }, [foldersList, initialValues, isMissingFolderId, sharedFolder])
```